src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/GCRefMapBuilder.cs

```diff
@@ -290,6 +289,26 @@ private void GcScanValueType(TypeDesc type, ArgDestination argDest, int delta, C
             }
         }
 
+        private void FindByRefPointerOffsetsInByRefLikeObject(TypeDesc type, ArgDestination argDest, int delta, CORCOMPILE_GCREFMAP_TOKENS[] frame)
+        {
+            if (type.IsByReferenceOfT || type.IsByRef)
+            {
+                argDest.GcMark(frame, delta, interior: true);
+                return;
+            }
+
+            foreach (FieldDesc field in type.GetFields())
+            {
+                if (!field.IsLiteral && 
```

Comment:
Literal fields are a subset of static fields. We could just drop this to have less code.

Reply:
Fixed in 3rd commit, thanks.